docs: add note on React canary version to View transitions guide (#92315)
vercel/next.js: docs/01-app/02-guides/view-transitions.mdx

export default nextConfig
```

> App Router uses [React canary releases](https://react.dev/blog/2023/05/03/react-canaries) built-in, which include all the stable React 19 changes, as well as newer features being validated in frameworks, such as `ViewTransition`. You do not need to install `react@canary` yourself.

Reviewer: ```suggestion
App Router uses [React canary releases](https://react.dev/blog/2023/05/03/react-canaries), which contain all stable React 19 changes as well as newer features like ViewTransition. You do not need to install react@canary yourself.
```

import { ViewTransition } from 'react'
```

> > [!NOTE]

Reviewer: Let's add this after the config snippet, line 39-ish

No need to be a note I think. 

And we can use the note from the landing page, /docs

```suggestion
App Router uses [React canary releases](https://react.dev/blog/2023/05/03/react-canaries) built-in, which include all the stable React changes, and newer features, such as `ViewTransitions`. You do not need to install `react@canary` yourself.
```

Author: How about now?
